app/dao/notifications_dao.py

```diff
@@ -96,7 +96,7 @@ def country_records_delivery(phone_prefix):
 def _update_notification_status(notification, status, provider_response=None):
     status = _decide_permanent_temporary_failure(current_status=notification.status, status=status)
     notification.status = status
-    if status == NOTIFICATION_TECHNICAL_FAILURE and provider_response:
```

Comment:
This condition has been moved to the SNS code now that SES fills this field as well